### PR TITLE
Update array-manager.md

### PR DESCRIPTION
**Pull request by RakeshJesadiya**
Added Missing methods from the ArrayManager class and added one more example of the newly added functions.

## Purpose of this pull request

This pull request (PR) adds the missing functions on the ArrayManager class.

## Affected DevDocs pages

https://devdocs.magento.com/guides/v2.3/extension-dev-guide/framework/array-manager.html
https://devdocs.magento.com/guides/v2.4/extension-dev-guide/framework/array-manager.html

### CODE REVIEW

File: src/guides/v2.3/extension-dev-guide/framework/array-manager.md [hunk 1]
```diff
@@ -13,11 +13,16 @@ The library is primarily used to handle data from UI components within [DataProv
 |Method|Description|
 |--- |--- |
 | `exists` | Checks if the node exists in a given associative array |
-| `get` | Returns the value of the key (or node) at the end of the path, `null` is returned if the node hasn't been found. |
+| `find` | Finds node in nested array and saves its index and parent node reference |
```
Comment (keharper) [inline src/guides/v2.3/extension-dev-guide/framework/array-manager.md:16]:
```suggestion
| `find` | Finds a node in a nested array and saves its index and parent node reference |
```

Add a period to the end of each line in this table.

File: src/guides/v2.3/extension-dev-guide/framework/array-manager.md [hunk 2]
```diff
@@ -13,11 +13,16 @@ The library is primarily used to handle data from UI components within [DataProv
...
+| `findPaths` | Get matching paths for elements with specified indexes |
+| `get` | Returns the value of the key (or node) at the end of the path, `null` is returned if the node hasn't been found |
```
Comment (keharper) [inline src/guides/v2.3/extension-dev-guide/framework/array-manager.md:18]:
```suggestion
| `get` | Returns the value of the key (or node) at the end of the path. `null` is returned if the node could not be found. |
```

File: src/guides/v2.3/extension-dev-guide/framework/array-manager.md [hunk 3]
```diff
@@ -13,11 +13,16 @@ The library is primarily used to handle data from UI components within [DataProv
...
 | `move` | Moves a value from one path to another |
+| `merge` | Merge value with node and return modified data |
```
Comment (keharper) [inline src/guides/v2.3/extension-dev-guide/framework/array-manager.md:20]:
```suggestion
| `merge` | Merges a value with a node and return the modified data. |
```

File: src/guides/v2.3/extension-dev-guide/framework/array-manager.md [hunk 4]
```diff
@@ -13,11 +13,16 @@ The library is primarily used to handle data from UI components within [DataProv
...
+| `populate` | Populate nested array if possible and needed |
```
Comment (keharper) [inline src/guides/v2.3/extension-dev-guide/framework/array-manager.md:21]:
```suggestion
| `populate` | Populates a nested array, if possible and necessary. |
```

File: src/guides/v2.3/extension-dev-guide/framework/array-manager.md [hunk 5]
```diff
@@ -13,11 +13,16 @@ The library is primarily used to handle data from UI components within [DataProv
 |Method|Description|
 |--- |--- |
 | `exists` | Checks if the node exists in a given associative array |
-| `get` | Returns the value of the key (or node) at the end of the path, `null` is returned if the node hasn't been found. |
+| `find` | Finds node in nested array and saves its index and parent node reference |
+| `findPaths` | Get matching paths for elements with specified indexes |
```
Comment (keharper) [inline src/guides/v2.3/extension-dev-guide/framework/array-manager.md:17]:
```suggestion
| `findPaths` | Gets matching paths for elements with specified indexes. |
```

File: src/guides/v2.3/extension-dev-guide/framework/array-manager.md [hunk 6]
```diff
@@ -13,11 +13,16 @@ The library is primarily used to handle data from UI components within [DataProv
...
+| `get` | Returns the value of the key (or node) at the end of the path, `null` is returned if the node hasn't been found |
 | `move` | Moves a value from one path to another |
+| `merge` | Merge value with node and return modified data |
+| `populate` | Populate nested array if possible and needed |
 | `remove` | Removes node and returns modified array |
 | `replace` | Updates the existing nodes and returns the modified array |
 | `set` | Set value into node and returns modified data |
+| `slicePath` | Retrieve slice of specified path |
```
Comment (keharper) [inline src/guides/v2.3/extension-dev-guide/framework/array-manager.md:25]:
```suggestion
| `slicePath` | Retrieves a slice of the specified path. |
```

File: src/guides/v2.3/extension-dev-guide/framework/array-manager.md [hunk 7]
```diff
@@ -13,11 +13,16 @@ The library is primarily used to handle data from UI components within [DataProv
...
-| `get` | Returns the value of the key (or node) at the end of the path, `null` is returned if the node hasn't been found. |
+| `find` | Finds node in nested array and saves its index and parent node reference |
+| `findPaths` | Get matching paths for elements with specified indexes |
+| `get` | Returns the value of the key (or node) at the end of the path, `null` is returned if the node hasn't been found |
 | `move` | Moves a value from one path to another |
+| `merge` | Merge value with node and return modified data |
+| `populate` | Populate nested array if possible and needed |
 | `remove` | Removes node and returns modified array |
 | `replace` | Updates the existing nodes and returns the modified array |
 | `set` | Set value into node and returns modified data |
```
Comment (keharper) [inline src/guides/v2.3/extension-dev-guide/framework/array-manager.md:24]:
```suggestion
| `set` | Sets the value into a node and returns modified data. |
```

File: src/guides/v2.3/extension-dev-guide/framework/array-manager.md [hunk 8]
```diff
@@ -13,11 +13,16 @@ The library is primarily used to handle data from UI components within [DataProv
...
 |--- |--- |
 | `exists` | Checks if the node exists in a given associative array |
-| `get` | Returns the value of the key (or node) at the end of the path, `null` is returned if the node hasn't been found. |
+| `find` | Finds node in nested array and saves its index and parent node reference |
+| `findPaths` | Get matching paths for elements with specified indexes |
+| `get` | Returns the value of the key (or node) at the end of the path, `null` is returned if the node hasn't been found |
 | `move` | Moves a value from one path to another |
+| `merge` | Merge value with node and return modified data |
+| `populate` | Populate nested array if possible and needed |
 | `remove` | Removes node and returns modified array |
```
Comment (keharper) [inline src/guides/v2.3/extension-dev-guide/framework/array-manager.md:22]:
```suggestion
| `remove` | Removes a node and returns the modified array. |
```

File: src/guides/v2.3/extension-dev-guide/framework/array-manager.md [hunk 9]
```diff
@@ -129,3 +134,13 @@ if ($this->arrayManager->get('response/status', $data) === 'OK') {
 
 ...
 ```
+
+You can use the  `Magento\Framework\Stdlib\ArrayManager` library to populate array from the given path:
```
Comment (keharper) [inline src/guides/v2.3/extension-dev-guide/framework/array-manager.md:138]:
```suggestion
You can use the  `Magento\Framework\Stdlib\ArrayManager` library to populate an array from the given path:
```

**Review by keharper — CHANGES_REQUESTED** [review]
@RakeshJesadiya This PR will require technical review. Adobe/Magento will be shut down from Dec 24 to Jan 4, so it'll be a while until the PR can be processed further.

**Comment by RakeshJesadiya** [comment]
@keharper  I have updated changes as per your suggestion.

**Review by dobooth — APPROVED** [review]

**Review by BarnyShergold — APPROVED** [review]

**Comment by dobooth** [comment]
running tests

**Pull request merged**